bsdfdisk/bsdfdisk-lib.pl

```diff
@@ -1,468 +1,538 @@
-my $ex = $?;
-&unlink_file($temp);
-return $ex ? $out : undef;
-}
-
-# create_filesystem(&disk, &slice, &part, &fs-details)
-# Creates a new filesystem, and returns undef on success or the error output
-# on failure.
-sub create_filesystem
-{
-my ($disk, $slice, $part, $newfs) = @_;
-my $cmd = &get_create_filesystem_command($disk, $slice, $part, $newfs);
-my $out = &backquote_logged("$cmd 2>&1 </dev/null");
-return $? ? $out : undef;
-}
-
-# get_create_filesystem_command(&disk, &slice, &part, &fs-details)
-# Returns the command to create a new filesystem on some partition
-sub get_create_filesystem_command
-{
-my ($disk, $slice, $part, $newfs) = @_;
-my @cmd = "newfs";
-push(@cmd, "-m", $newfs->{'free'}) if ($newfs->{'free'} ne '');
-push(@cmd, "-t") if ($newfs->{'trim'});
-push(@cmd, "-L", quotemeta($newfs->{'label'})) if ($newfs->{'label'} ne '');
-push(@cmd, $part ? $part->{'device'} : $slice->{'device'});
-return join(" ", @cmd);
-}
-
-# check_filesystem(&disk, &slice, &part)
-# Checks the filesystem on some partition,  and returns undef on success or
-# the error output on failure.
-sub check_filesystem
-{
-my ($disk, $slice, $part) = @_;
-my $cmd = &get_check_filesystem_command($disk, $slice, $part);
-my $out = &backquote_logged("$cmd 2>&1 </dev/null");
-return $? ? $out : undef;
-}
-
-# get_check_filesystem_command(&disk, &slice, &part)
-# Returns the command to check a filesystem on some partition
-sub get_check_filesystem_command
-{
-my ($disk, $slice, $part) = @_;
-my $dev = $part ? $part->{'device'} : $slice->{'device'};
-my @cmd = "fsck";
-my @st = &fdisk::device_status($dev);
-if (!@st) {
-	# Assume UFS type
-	push(@cmd, "-t", "ufs");
-	}
-push(@cmd, $dev);
-return join(" ", @cmd);
-}
-
-# show_filesystem_buttons(hiddens, &status, &part-or-slice)
-# Show buttons to create a filesystem on a partition or slice
-sub show_filesystem_buttons
-{
-my ($hiddens, $st, $object) = @_;
-print &ui_buttons_row(
-	"newfs_form.cgi", $text{'part_newfs'}, $text{'part_newfsdesc'},
-	$hiddens);
-
-if (!@$st || $st->[1] ne 'swap') {
-	print &ui_buttons_row(
-		"fsck.cgi", $text{'part_fsck'}, $text{'part_fsckdesc'},
-		$hiddens);
-	}
-
-if (!@$st) {
-	if ($object->{'type'} eq 'swap' || $object->{'type'} eq '82') {
-		print &ui_buttons_row("../mount/edit_mount.cgi",
-		    $text{'part_newmount2'}, $text{'part_mountmsg2'},
-		    &ui_hidden("newdev", $object->{'device'}).
-		    &ui_hidden("type", "swap"));
-		}
-	else {
-		print &ui_buttons_row("../mount/edit_mount.cgi",
-		    $text{'part_newmount'}, $text{'part_mountmsg'},
-		    &ui_hidden("newdev", $object->{'device'}).
-		    &ui_hidden("type", "ufs"),
-		    &ui_textbox("newdir", undef, 20));
-		}
-	}
-}
-
-1;
+# Functions for FreeBSD disk management
+
+use strict;
+use warnings;
+no warnings 'redefine';
+no warnings 'uninitialized';
+BEGIN { push(@INC, ".."); };
+use WebminCore;
+&init_config();
+&foreign_require("mount");
+&foreign_require("fdisk");

```

Comment:
@karmantyu, I reverted back this change to use `fdisk` module, because using `mount::device_status` seems incorrect, simply because `device_status` sub isn't available in `mount` module.